src/java.desktop/unix/classes/sun/awt/X11/XWindow.java

```diff
@@ -303,15 +303,13 @@ static long getParentWindowID(Component target) {
         Component temp = target.getParent();
         final ComponentAccessor acc = AWTAccessor.getComponentAccessor();
         ComponentPeer peer = acc.getPeer(temp);
-        while (!(peer instanceof XWindow))
+        while (!(peer instanceof XWindow window))
         {
             temp = temp.getParent();
             peer = acc.getPeer(temp);
         }
 
-        if (peer != null && peer instanceof XWindow)
-            return ((XWindow)peer).getContentWindow();
-        else return 0;
+        return window.getContentWindow();
```

Comment:
Is the branch where 0 was returned impossible?

Reply:
Yes. It was impossible.
Only way out of this cycle is when `peer instanceof XWindow` is `true`
https://github.com/openjdk/jdk/blob/0c7a4b8aa8bb672e87aae7090494719db018b9b1/src/java.desktop/unix/classes/sun/awt/X11/XWindow.java#L306-L310

Reply:
Thanks, I just wanted to confirm my own understanding.